build/gulpfile.sql.js

```diff
@@ -96,50 +94,39 @@ const formatStagedFiles = () => {
-		let runtime = p.runtimeId;
-		// fix path since it won't be correct
-		config.installDirectory = path.join(__dirname, '../extensions/mssql/src', config.installDirectory);
-		let installer = new serviceDownloader(config);
-		installer.eventEmitter.onAny((event, ...values) => {
-			console.log(`ServiceDownloader Event : ${event}${values && values.length > 0 ? ` - ${values.join(' ')}` : ''}`);
-		});
-		let serviceInstallFolder = installer.getInstallDirectory(runtime);
-		console.log('Cleaning up the install folder: ' + serviceInstallFolder);
-		return del(serviceInstallFolder + '/*').then(() => {
-			console.log('Installing the service. Install folder: ' + serviceInstallFolder);
-			return installer.installService(runtime);
-		}, delError => {
-			console.log('failed to delete the install folder error: ' + delError);
-		});
-	});
-}
-
-gulp.task('install-sqltoolsservice', () => {
-	return installService();
-});
-
-gulp.task('install-ssmsmin', () => {
-	const config = require('../extensions/admin-tool-ext-win/config.json');
-	const runtime = 'Windows_64'; // admin-tool-ext is a windows only extension, and we only ship a 64 bit version, so locking the binaries as such
+async function installService(configPath, runtimId) {
+	const absoluteConfigPath = require.resolve(configPath);
+	const config = require(absoluteConfigPath);
+	const runtime = runtimId || (await platform.getCurrent()).runtimeId;
 	// fix path since it won't be correct
-	config.installDirectory = path.join(__dirname, '..', 'extensions', 'admin-tool-ext-win', config.installDirectory);
+	config.installDirectory = path.join(path.dirname(absoluteConfigPath), config.installDirectory);
+	console.log('install diectory', config.installDirectory);
 	let installer = new serviceDownloader(config);
 	installer.eventEmitter.onAny((event, ...values) => {
 		console.log(`ServiceDownloader Event : ${event}${values && values.length > 0 ? ` - ${values.join(' ')}` : ''}`);
 	});
-	const serviceInstallFolder = installer.getInstallDirectory(runtime);
-	const serviceCleanupFolder = path.join(serviceInstallFolder, '..');
-	console.log('Cleaning up the install folder: ' + serviceCleanupFolder);
-	return del(serviceCleanupFolder + '/*').then(() => {
-		console.log('Installing the service. Install folder: ' + serviceInstallFolder);
-		return installer.installService(runtime);
-	}, delError => {
-		console.log('failed to delete the install folder error: ' + delError);
-	});
-});
+	let serviceInstallFolder = installer.getInstallDirectory(runtime);
+	console.log('Cleaning up the install folder: ' + serviceInstallFolder);
+	try {
+		await util.rimraf(serviceInstallFolder)();
+	} catch (e) {
+		console.error('failed to delete the install folder error: ' + e);
+		throw e;
+	}
+	await installer.installService(runtime);
+	let stat;
+	for (const file of config.executableFiles) {
+		try {
+			stat = await fs.stat(path.join(serviceInstallFolder, file));
+		} catch (e) { }
+	}
+
+	assert(stat);
```

Comment:
If the first one fails but the subsequent ones don't then this won't fail the build. Shouldn't this check be inside the try?

Reply:
We don't care if every file exists, we only care if at least 1 file exists